executor/envd/service/process_service.py

```
@@ -0,0 +1,799 @@
+        if selector.HasField("pid"):
+            return selector.pid
+        elif selector.HasField("tag"):
+            return self.tagged_processes.get(selector.tag)
+        return None
+
+    def list_processes(self):
+        """List all managed processes"""
+        return [
+            (pid, proc, tag if tag and self.tagged_processes.get(tag) == pid else None)
+            for pid, proc in self.processes.items()
+            for tag in [None] + list(self.tagged_processes.keys())
```

Ruff (0.14.11)

Consider `[None, *list(self.tagged_processes.keys())]` instead of concatenation

Replace with `[None, *list(self.tagged_processes.keys())]`

(RUF005)